packages/flutter/lib/src/widgets/text_editing_actions.dart

```diff
@@ -0,0 +1,558 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'framework.dart';
+import 'text_editing_action.dart';
+import 'text_editing_intents.dart';
+
+/// An [Actions] widget that handles the default text editing behavior for
+/// Flutter on the current platform.
+///
+/// This default behavior can be overridden by placing an [Actions] widget lower
+/// in the widget tree than this. See [TextEditingShortcuts] for an example of
+/// remapping keyboard keys to an existing text editing [Intent].
+///
+/// {@tool snippet}
+///
+/// This example shows how to use an additional [Actions] widget to override
+/// the left arrow key [Intent] and make it move the cursor to the right
+/// instead.
+///
+/// ```dart
+/// final TextEditingController controller = TextEditingController(
+///   text: "Try using the keyboard's arrow keys and notice that left moves right.",
+/// );
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Center(
+///       child: Actions(
+///         actions: <Type, Action<Intent>>{
+///           ArrowLeftTextIntent: TextEditingAction<ArrowLeftTextIntent>(
+///             onInvoke: (ArrowLeftTextIntent intent, EditableTextState editableTextState) {
+///               editableTextState.renderEditable.moveSelectionRight(SelectionChangedCause.keyboard);
+///             },
+///           ),
+///         },
+///         child: TextField(
+///           controller: controller,
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [TextEditingShortcuts], which maps keyboard keys to many of the
+///     [Intent]s that are handled here.
+class TextEditingActions extends StatelessWidget {
+  /// Creates an instance of TextEditingActions.
+  TextEditingActions({
+    Key? key,
+    Map<Type, Action<Intent>>? additionalActions,
+    required this.child,
+  }) : additionalActions = additionalActions ?? <Type, Action<Intent>>{},
+       super(key: key);
+
+  /// The child [Widget] of TextEditingActions.
+  final Widget child;
+
+  /// The actions to be merged with the default text editing actions.
+  ///
+  /// The default text editing actions will override any conflicting keys in
+  /// additionalActions. To override the default text editing actions, use an
+  /// [Actions] Widget in the tree below this Widget.
+  final Map<Type, Action<Intent>> additionalActions;
+
+  static final TextEditingAction<AltArrowDownTextIntent> _altArrowDownTextAction = TextEditingAction<AltArrowDownTextIntent>(
```

Comment:
By making the actions simply map one-to-one to intents, which in turn, just map to keys, I think we're discarding most of the value of the key-to-action lookup system. Why bother with intents?

Reply:
I may be missing some nuance of the architecture, but wouldn't it be ideal to name intents based on the intent of the user?

For example ctrl+backspace may be the shortcut while the intent is delete line. Or shift+left may be the shortcut while the intent is extend selection left.

Then depending on the platform, we'd have different shortcut-intent mappings.